app/controllers/settings_controller.rb

```diff
@@ -3,26 +3,27 @@
 class SettingsController < ApplicationController
   include Europeana::Styleguide
 
-  ##
-  # @todo move this into two separate actions, one for GET, one for PUT, with
-  #  a resourceful controller and routes, per http://guides.rubyonrails.org/routing.html#singular-resources
+  # GET language
+  # @todo move this into a resourceful controller and routes for Settings::Languages
+  #   per http://guides.rubyonrails.org/routing.html#singular-resources
   def language
-    if request.put?
-      locale_changed = set_locale_from_param
+    respond_to do |format|
+      format.html
+    end
+  end
 
-      respond_to do |format|
-        format.html do
-          render action: :language, status: status_from_flash
-        end
-        format.json do
-          render json: { success: !flash.key?(:alert), refresh: locale_changed,
-                         message: flash.key?(:alert) ? flash.now[:alert] : flash.now[:notice] },
-                 status: status_from_flash
-        end
+  # PUT language
+  def update_language
```

Comment:
Assignment Branch Condition size for update_language is too high. [22.05/15]<br>Method has too many lines. [11/10]